presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkQueryExecutionFactory.java

```diff
@@ -728,13 +746,16 @@ private PrestoSparkQueryExecution(
             this.transactionManager = requireNonNull(transactionManager, "transactionManager is null");
             this.pagesSerde = requireNonNull(pagesSerde, "pagesSerde is null");
             this.executionExceptionFactory = requireNonNull(executionExceptionFactory, "executionExceptionFactory is null");
+            this.queryTimeout = requireNonNull(queryTimeout, "queryTimeout is null");
             this.queryStatusInfoOutputPath = requireNonNull(queryStatusInfoOutputPath, "queryStatusInfoOutputPath is null");
             this.queryDataOutputPath = requireNonNull(queryDataOutputPath, "queryDataOutputPath is null");
         }
 
         @Override
         public List<List<Object>> execute()
         {
+            queryCompletionDeadline = System.currentTimeMillis() + queryTimeout.toMillis();
```

Comment:
`queryCompletionDeadline` can be initialized immediately after `queryStateTimer` is created.

Reply:
Done